client/mm/event_log.go

```diff
@@ -0,0 +1,672 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package mm
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/encode"
+	"go.etcd.io/bbolt"
+)
+
+type dexOrderEvent struct {
+	ID           string                     `json:"id"`
+	Rate         uint64                     `json:"rate"`
+	Qty          uint64                     `json:"qty"`
+	Sell         bool                       `json:"sell"`
+	Transactions []*asset.WalletTransaction `json:"transactions"`
+}
+
+type cexOrderEvent struct {
+	ID          string `json:"id"`
+	Rate        uint64 `json:"rate"`
+	Qty         uint64 `json:"qty"`
+	Sell        bool   `json:"sell"`
+	BaseFilled  uint64 `json:"baseFilled"`
+	QuoteFilled uint64 `json:"quoteFilled"`
+}
+
+type depositEvent struct {
+	Transaction *asset.WalletTransaction `json:"transaction"`
+	AssetID     uint32                   `json:"assetID"`
+	CEXCredit   uint64                   `json:"cexCredit"`
+}
+
+type withdrawalEvent struct {
+	ID          string                   `json:"id"`
+	AssetID     uint32                   `json:"assetID"`
+	Transaction *asset.WalletTransaction `json:"transaction"`
+	CEXDebit    uint64                   `json:"cexDebit"`
+}
+
+// MarketMakingEvent represents an action that a market making bot takes.
+type MarketMakingEvent struct {
+	ID         uint64 `json:"id"`
+	TimeStamp  int64  `json:"timeStamp"`
```

Comment:
Not to be picky, but I'm seeing inconsistent case usage for `timestamp` vs. `timeStamp`. I consider it one word, and I think we generally don't capitalize the `Stamp` part (or we just shorten it to `stamp` a lot of places).